### ISSUE TITLE
azurerm_automation_schedule altering start time based on timezone

### ISSUE THREAD

**Issue by maistro-mw** [hashicorp/terraform-provider-azurerm#17641]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.2

### AzureRM Provider Version

3.13.0

### Affected Resource(s)/Data Source(s)

azurerm_automation_schedule

### Terraform Configuration Files

```hcl
terraform {
  
  required_version = ">=1.2"

  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "~>3.3"
    }
    local = {
      source = "hashicorp/local"
    }
  }
  
  backend "azurerm" {
  }
}

provider "azurerm" {
  features {}
}

resource "azurerm_automation_schedule" "schedule" {
  name                    = "PowerBI-Embedded-Suspend"
  resource_group_name     = "Resource-Group"
  automation_account_name = "Automation-Account"
  frequency               = Week
  interval                = 1
  start_time              = "${substr(timeadd(timestamp(), "24h"), 0, 10)}T22:00:00Z"
  timezone                = "Europe/London"
  week_days               = ["Monday", "Tuesday", "Wednesday", "Thursday", "Friday"]
  description             = "Pause the PowerBI Embedded instance"
}
```

### Debug Output/Panic Output

```shell
No debug logs
```

### Expected Behaviour

The schedule is entered as T22:00Z with timezone Europe/London, so I expect the schedule to be created for 10pm in the UK timezone (which currently is summer time so UTC+1).
The expected behaviour (which would occur is manually creating in Azure) would be for the schedule to run at 10pm - observing the summer time changes - so it is always at 10pm.

### Actual Behaviour

The schedule is created as 11pm in the UK timezone.  The apply operation is adding the +1 but keeping the timezone setting.
This means the schedule will run at 11pm in the summer time but if I were to run the apply again out of summer time it would reset to 10pm.

**Comment by maistro-mw** [comment]
Hi @wuxu92 thanks for your reply.  I tried using +01:00 instead of Z but that still put the wrong time in unless I left the timezone property blank (thus defaulting to UTC).  This removed the Timezone setting in the Azure resource which meant it would not cope with the summer time changes.
It would be good if this could be addressed by the Terraform team - prior to that I will manually adjust my definition to result in the correct time being deployed.

**Comment by maistro-mw** [comment]
OK, thanks.  I will try again - I was not getting the same result

**Comment by maistro-mw** [comment]
@wuxu92 Thanks - that appears to get the correct time in now.  Will see what happens when summer time ends as I guess I will need to change to +00:00 in my terraform config

**Comment by DMunkov** [comment]
I think the best option here would be to completely ignore time-offset in favour of timezone if specified.

**Comment by simaotwx** [comment]
There is a solution for this on SO: https://stackoverflow.com/questions/61762648/terraform-azurerm-schedule-start-time-always-resets-on-new-deploys

**Comment by MichalSino** [comment]
Hi @wuxu92,

 I have the same problem. I'm trying to set schedule in local time (fixed) 
`start_time     = "2023-11-23T18:20:31Z"`
`timezone       = "Europe/Warsaw"`

and in portal I have:
![image](https://github.com/hashicorp/terraform-provider-azurerm/assets/74066196/ca54e0a3-ecc8-4820-b3f0-d7b899001a34)

When I do it in portal I'm entering local time, so I think the is some calculation of start_time and timezone.

Can it be fixed?

**Comment by MichalSino** [comment]
Ok, I understand.

But that's why I ask to possibility of changing it, because I need to remember if there is summer time or winter time. If it were not be calculated and writed the same as I entered, it would be better. Do you know  it is done in api side or in provider side?

**Comment by akmalharith** [comment]
I don't really understand why accept tz database `timezone` input but actually require a tz-aware RFC3339 input in `start_time`..

**Comment by mpjtaylor** [comment]
Agreed, if you do it in the portal the time is set correctly with the timezone, but via terraform you get a totally different time when using timezone and start_time.

**Comment by Netkracker** [comment]
any news on this one ? still seems to happen with azurerm [v4.57.0](https://github.com/hashicorp/terraform-provider-azurerm/releases/tag/v4.57.0)
